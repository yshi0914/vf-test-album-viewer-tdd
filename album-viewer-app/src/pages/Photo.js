import React, { useState } from "react";
import PhotoList from "../components/PhotoList";
const Photo = (props) => {
  const [photos, setPhotos] = useState([]);

  // load all the albums
  fetch(
    `https://jsonplaceholder.typicode.com/photos?albumId=${props.match.params.id}`
  )
    .then((response) => response.json())
    .then((json) => {
      setPhotos(json);
    });

  let list = <div>Fetching data ...</div>;
  if (photos.length === 0) {
    list = <h4>Waiting for the data...</h4>;
  }
  const albumName = new URLSearchParams(props.location.search).get("albumName");

  if (photos && photos.length > 0) {
    list = <PhotoList data={photos} />;
  } else {
    list = <h4>Loading ....</h4>;
  }

  return (
    <div>
      <h1>Album Name : {albumName}</h1>
      {list}
    </div>
  );
};

export default Photo;
