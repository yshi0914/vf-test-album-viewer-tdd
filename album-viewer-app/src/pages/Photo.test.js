import React from "react";
import Photo from "./Photo";
import { shallow, mount } from "enzyme";
// Pass the url params and the album id into photo component
// Check if the album name is displayed correctly
describe("Photo component after mounting", () => {
  let wrapper;
  test("render the album name of this photo component", () => {
    wrapper = mount(
      <Photo
        match={{ params: { id: 1 } }}
        location={{ search: { albumName: "A Great Album Name" } }}
      />
    );
    expect(wrapper.find("h1").text()).toContain("A Great Album Name");
  });
});

// Snapshot Testing for Photo component
describe("AlbumList component snapshot testing", () => {
  test("matches the snapshot", () => {
    const component = shallow(
      <Photo
        match={{ params: { id: 1 } }}
        location={{ search: { albumName: "A Great Album Name" } }}
      />
    );
    expect(component.getElements()).toMatchSnapshot();
  });
});
