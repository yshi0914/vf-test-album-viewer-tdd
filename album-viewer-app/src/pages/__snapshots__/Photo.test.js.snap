// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumList component snapshot testing matches the snapshot 1`] = `
Array [
  <div>
    <h1>
      Album Name : 
      A Great Album Name
    </h1>
    <h4>
      Loading ....
    </h4>
  </div>,
]
`;
